node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -193,19 +201,29 @@ object WebApi {
       usePreStateHash: Boolean
   )
 
-  final case class DataRequest(
+  final case class DataAtNameRequest(
       // For simplicity only one Unforgeable name is allowed
       // instead of the whole RhoExpr (proto Par)
       name: RhoUnforg,
       // Number of blocks in the past to search for data
       depth: Int
   )
 
-  final case class DataResponse(
+  final case class DataAtParRequest(
+      blockHash: String,
+      name: RhoUnforg,
+      usePreStateHash: Boolean
+  )
+
+  final case class DataAtNameResponse(
       exprs: List[RhoExprWithBlock],
       length: Int
   )
 
+  final case class DataAtParResponse(
+      pars: Seq[Par]
```

Comment:
Existing WebAPI methods returns its own representation of `Par` type which is `RhoExpr`. So new method should return `Seq[RhoExpr]` and `LightBlockInfo` object for the block on which data is returned.

If user didn't specify block hash as input, last finalized block will be used and LightBlockInfo will give user info for what state he is getting the data.

Reply:
We should reuse `ExploratoryDeployResponse` type which has the same structure. But renaming it to `RhoDataResponse`.